transport_nantes/photo/views.py

```diff
@@ -43,9 +50,38 @@ def post(self, request, *args, **kwargs):
         else:
             return self.form_invalid(form)
 
+    def form_valid(self, form):
+        """
+        Override the form_valid method to add the user's submission to session
+        """
+        self.object = form.save()
+        encoded_object_id = make_timed_token(
+            str(self.object.id), minutes=60*24*30)
+        self.success_url += f"?submission={encoded_object_id}"
+        return HttpResponseRedirect(self.get_success_url())
+
 
 class Confirmation(TemplateView):
     """
     Confirmation page after successful submit of a PhotoEntry
     """
     template_name = 'photo/confirmation.html'
+
+    def get(self, request, *args, **kwargs):
+        logger.info(f"Confirmation.get() from {request.user}")
+        return super().get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        last_submitted_photo = None
+        submission_token = self.request.GET.get("submission")
+        photo_id, _ = token_valid(submission_token)
+
+        if photo_id.isnumeric():
```

Comment:
Why `is_numeric()`?  If it's not valid, you'll get `None`.

This is a weird abuse of the timed token, though.  If you need to recover an int and not a string, then just pass that.  If you don't need the string, pass an empty string.  Casting back and forth presents too many opportunities for errors or things going wrong that lead to a string that can be converted to an int and you get what you asked for but not what you wnated.

Reply:
Fixed, passed through the int_key instead

Reply:
You also pass the string as `None` instead of empty ("").  But the validation function returns `None` on invalid.  So even though you don't look at the string on return (not looking at error return?), you can't distinguish between a valid token with `int_key = 0` and an invalid token.